src/processors/modulation/Flanger.cpp

```diff
@@ -0,0 +1,298 @@
+
+        auto phaseShiftLFO = [this, numSamples, filterIndex = 0] (float* lfoIn, float* lfoOut) mutable
+        {
+            // trying to use more of the Hilbert filters than we have available!
+            jassert (filterIndex < (int) std::size (hilbertFilter));
+
+            auto& filter = hilbertFilter[filterIndex++];
+            for (int n = 0; n < numSamples; ++n)
+                std::tie (lfoIn[n], lfoOut[n]) = filter.process (lfoIn[n]);
+        };
+
+        auto* modOutData = modOutBuffer.getReadPointer (0);
+
+        FloatVectorOperations::copy (LFOData[0][0].data(), modOutData, numSamples);
+        phaseShiftLFO (LFOData[0][0].data(), LFOData[0][1].data());
+        FloatVectorOperations::copy (LFOData[1][0].data(), LFOData[0][1].data(), numSamples);
+        phaseShiftLFO (LFOData[1][0].data(), LFOData[1][1].data());
+    }
+    else
+    {
+        auto rate = rateLow * std::pow (rateHigh / rateLow, *rateParam);
+        for (int ch = 0; ch < 2; ++ch)
+        {
+            for (int i = 0; i < delaysPerChannel; ++i)
+            {
+                LFOs[ch][i].setFrequency (rate);
+            }
+
+            for (int i = 0; i < delaysPerChannel; ++i)
+            {
+                auto* data = LFOData[ch][i].data();
+                for (int n = 0; n < numSamples; ++n)
+                    data[n] = LFOs[ch][i].processSample();
+            }
+        }
+
+        auto* modOutData = modOutBuffer.getWritePointer (0);
+        FloatVectorOperations::copy (modOutData, LFOData[0][0].data(), numSamples);
+    }
+}
+
+template <typename DelayArrType>
+void Flanger::processFlanger (AudioBuffer<float>& buffer, DelayArrType& delay)
+{
+    jassert (buffer.getNumChannels() == 2); // always processing in stereo
+    const auto numSamples = buffer.getNumSamples();
+
+    for (int ch = 0; ch < 2; ++ch)
+    {
+        for (int i = 0; i < delaysPerChannel; ++i)
+            delay[ch][i].setFilterFreq (10000.0f);
+
+        auto fbAmount = std::sqrt (fbParam->getCurrentValue());
+        if constexpr (std::is_same_v<DelayArrType, decltype (lofiDelay)>)
+            fbAmount *= 0.4f;
+        else
+            fbAmount *= 0.5f;
+
+        fbSmooth[ch].setTargetValue (fbAmount);
+
+        delaySmoothSamples[ch].setTargetValue (delayMs * fs * delayAmountParam->getCurrentValue());
+        delayOffsetSmoothSamples[ch].setTargetValue (delayMs * fs * delayOffsetParam->getCurrentValue());
+
+        auto* x = buffer.getWritePointer (ch);
+        for (int n = 0; n < numSamples; ++n)
+        {
+            auto xIn = std::tanh (x[n] * 0.75f - feedbackState[ch]);
+
+            auto delayValue = delaySmoothSamples[ch].getNextValue();
+            auto delayOffsetValue = delayOffsetSmoothSamples[ch].getNextValue();
+
+            x[n] = 0.0f;
+            for (int i = 0; i < delaysPerChannel; ++i)
+            {
+                float delayAmt = delayValue + delayOffsetValue * 0.5f * (1.0f + 0.95f * LFOData[ch][i][(size_t) n]);
```

Comment:
With the way this is currently set up there will be no effect unless the `delayOffsetValue` != 0. Is that how its supposed to work?